pyproject.toml

```
@@ -12,17 +12,18 @@ authors = []
 python = "^3.10"
 ops = "^2.5.0"
 tenacity = "^8.2.2"
-pydantic = "^1.10.7"
 boto3 = "^1.28.23"
 # TODO: Remove insecure dependency https://github.com/canonical/mysql-operator/issues/246
 pyopenssl = "^23.2.0"
 
 [tool.poetry.group.charm-libs.dependencies]
 # data_platform_libs/v0/data_interfaces.py
 ops = ">=2.0.0"
+# data_platform_libs/v0/upgrade.py requires pydantic ^1.10
+# grafana_agent/v0/cos_agent.py requires pydantic <2
+pydantic = "^1.10, <2"
```

maybe I'm dumb, but isn't the caret requirement redundant with the `<2`
```suggestion
pydantic = "^1.10"
```

---

I believe @carlcsaposs-canonical is willing to represent 2 libraries here pinning pydantic to notice on the next update. You are right, technically it is redundant.